packages/sdk/src/application/ApplicationClient.ts

```diff
@@ -0,0 +1,314 @@
+          ...(await this.listApplications({
+            pageSize,
+            pageToken: data.nextPageToken,
+            enablePagination,
+            view,
+            ownerId,
+          })),
+        );
+      }
+
+      return Promise.resolve(applications);
+    } catch (err) {
+      return Promise.reject(err);
+    }
+  }
+
+  async getApplication({ applicationName }: GetApplicationRequest) {
+    try {
+      const additionalHeaders = getInstillAdditionalHeaders({});
+      const data = await this._client.get<GetApplicationResponse>(
+        `/${applicationName}`,
+        { additionalHeaders },
+      );
+
+      return Promise.resolve(data.app);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async createConversation({
+    ownerId,
+    appId,
+    payload,
+  }: CreateConversationRequest) {
+    try {
+      const additionalHeaders = getInstillAdditionalHeaders({});
+      const response = await this._client.post<Conversation>(
+        `/namespaces/${ownerId}/apps/${appId}/conversations`,
+        {
+          body: JSON.stringify(payload),
+          additionalHeaders,
+        },
+      );
+      return Promise.resolve(response);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async getPlaygroundConversation({
+    ownerId,
+    appId,
+  }: GetPlaygroundConversationRequest) {
+    try {
+      const additionalHeaders = getInstillAdditionalHeaders({});
+      const response = await this._client.get<{ conversation: Conversation }>(
+        `/namespaces/${ownerId}/apps/${appId}/ai_assistant_playground/conversation`,
+        { additionalHeaders },
+      );
+      return Promise.resolve(response.conversation);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async chat({
+    ownerId,
+    appId,
+    catalogId,
+    conversationUid,
+    message,
+    topK = 15,
+    namespaceId,
+  }: ChatRequest): Promise<Response> {
+    try {
+      const additionalHeaders = getInstillAdditionalHeaders({
+        requesterUid: namespaceId,
+      });
+      const response = await this._client.post<Response>(
+        `/namespaces/${ownerId}/apps/${appId}/chat`,
+        {
+          body: JSON.stringify({
+            catalog_id: catalogId,
+            conversation_uid: conversationUid,
+            message: message,
+            top_k: topK,
+          }),
+          additionalHeaders: {
+            ...additionalHeaders,
+            Accept: "text/event-stream",
+          },
+        },
+      );
+      return response;
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async listMessages({ ownerId, appId, conversationId }: ListMessagesRequest) {
+    try {
+      const additionalHeaders = getInstillAdditionalHeaders({});
+      const response = await this._client.get<{ messages: Message[] }>(
+        `/namespaces/${ownerId}/apps/${appId}/conversations/${conversationId}/messages?ifAll=true`,
+        { additionalHeaders },
```

Comment:
@thewbuk User should be able to freely choose whether they want to set up this query parameter, we should make it as parameter